packages/engine/src/interaction/components/InteractableComponent.ts

```diff
@@ -23,6 +23,59 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
 Ethereal Engine. All Rights Reserved.
 */
 
+import { EntityUUID } from '@etherealengine/ecs'
 import { defineComponent } from '@etherealengine/ecs/src/ComponentFunctions'
+import { NO_PROXY } from '@etherealengine/hyperflux'
+import matches from 'ts-matches'
 
-export const InteractableComponent = defineComponent({ name: 'InteractableComponent' })
+export const InteractableComponent = defineComponent({
+  name: 'InteractableComponent',
+  jsonID: 'EE_interactable',
+  onInit: () => {
+    return {
+      label: null as string | null,
+      callbacks: [] as Array<{
+        /**
+         * The function to call on the CallbackComponent of the targetEntity when the trigger volume is entered.
+         */
+        callbackID: null | string
+        /**
+         * empty string represents self
+         */
+        target: null | EntityUUID
+      }>
+    }
+  },
+
+  onSet: (entity, component, json) => {
+    if (!json) return
+    if (json.label) component.label.set(json.label)
+
+    // backwards compatibility
+    const callbackID = (json as any).callbackID ?? null
```

Comment:
we don't need backwards compat outside the context of this PR, so lets remove this

Reply:
removed